.github/scripts/Release.py

```diff
@@ -0,0 +1,580 @@
+    return int(
+        pathlib.PurePosixPath(
+            urllib.parse.urlparse(new_version_url).path).parts[-1])
+
+
+def raise_for_status(response):
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError as err:
+        if response.status_code >= 400 and response.status_code < 500:
+            raise requests.exceptions.HTTPError(
+                yaml.safe_dump(response.json(), allow_unicode=True)) from err
+        else:
+            raise
+    return response
+
+
+@uplink.response_handler(raise_for_status)
+class Zenodo(uplink.Consumer):
+    @uplink.returns.json
+    @uplink.get('deposit/depositions')
+    def get_depositions(self):
+        """Retrieves all depositions."""
+        pass
+
+    @uplink.returns.json
+    @uplink.get('deposit/depositions/{id}')
+    def get_deposition(self, id: uplink.Path):
+        """Retrieves a deposition by ID."""
+        pass
+
+    @uplink.returns.json(key='doi')
+    @uplink.get('deposit/depositions/{id}')
+    def get_doi(self, id: uplink.Path):
+        """Retrieves only the DOI of a deposition by ID."""
+        pass
+
+    @uplink.returns.json
+    @uplink.get('records/{id}')
+    def get_record(self, id: uplink.Path):
+        """Retrieves a published record by ID."""
+        pass
+
+    @uplink.returns.json(key=('metadata', 'relations', 'version', 0,
+                              'last_child', 'pid_value'),
+                         type=int)
+    @uplink.get('records/{id}')
+    def get_latest_version_id(self, record_id: uplink.Path(name='id')):
+        """Retrieves only the latest version ID of a record."""
+        pass
+
+    @uplink.response_handler(parse_new_version_response)
+    @uplink.post('deposit/depositions/{id}/actions/newversion')
+    def new_version(self, latest_version_id: uplink.Path(name='id')):
+        """Invoke the "New version" action on a deposition.
+
+        Returns:
+          The ID of the new version.
+        """
+        pass
+
+    @uplink.returns.json
+    @uplink.post('deposit/depositions/{id}/actions/publish')
+    def publish(self, id: uplink.Path):
+        """Invoke the "Publish" action on a deposition."""
+        pass
+
+    @uplink.json
+    @uplink.returns.json
+    @uplink.put('deposit/depositions/{id}')
+    def update_deposition(self, id: uplink.Path, **body: uplink.Body):
+        """Update the deposition with the metadata"""
+        pass
+
+    # @uplink.response_handler(parse_bucket_response)
+    # @uplink.get('deposit/depositions/{id}')
+    # def get_bucket_id(self, deposition_id: uplink.Path(name='id')):
+    #     """Retrieves only the file bucket ID of a deposition"""
+    #     pass
+
+    @uplink.returns.json
+    @uplink.put('files/{bucket_id}/{filename}')
+    def upload_file(self, bucket_id: uplink.Path, filename: uplink.Path,
+                    file: uplink.Body):
+        """Uploads a file to the bucket."""
+        pass
+
+
+def get_response_text(response):
+    return response.text
```

Comment:
Sorry, why is this wrapper needed? It seems like calling the wrapper is more code than getting the text

Reply:
It's a "response handler" function, used for `Github.render_markdown_raw`. Replaces with `operator.attrgetter`.